pkg/providers/instance/instance.go

```diff
@@ -212,26 +214,50 @@ func (p *Provider) newNetworkInterfaceForVM(vmName string, backendPools *loadbal
 		enableAcceleratedNetworking = true
 	}
 
-	return armnetwork.Interface{
+	nic := armnetwork.Interface{
 		Location: to.Ptr(p.location),
 		Properties: &armnetwork.InterfacePropertiesFormat{
 			IPConfigurations: []*armnetwork.InterfaceIPConfiguration{
 				{
 					Name: &vmName,
 					Properties: &armnetwork.InterfaceIPConfigurationPropertiesFormat{
-						Primary:                   to.Ptr(true),
-						PrivateIPAllocationMethod: to.Ptr(armnetwork.IPAllocationMethodDynamic),
+						Primary:                   lo.ToPtr(true),
+						PrivateIPAllocationMethod: lo.ToPtr(armnetwork.IPAllocationMethodDynamic),
 						Subnet: &armnetwork.Subnet{
 							ID: &p.subnetID,
 						},
 						LoadBalancerBackendAddressPools: ipv4BackendPools,
 					},
 				},
 			},
-			EnableAcceleratedNetworking: to.Ptr(enableAcceleratedNetworking),
-			EnableIPForwarding:          to.Ptr(true),
+			EnableAcceleratedNetworking: lo.ToPtr(enableAcceleratedNetworking),
+			EnableIPForwarding:          lo.ToPtr(true),
 		},
 	}
+	networkPlugin := options.FromContext(ctx).NetworkPlugin 
+	networkPluginMode := options.FromContext(ctx).NetworkPluginMode
+	if networkPlugin == consts.NetworkPluginAzure && networkPluginMode != consts.PodNetworkTypeOverlay {
+		// NOTE: We don't need to set LoadBalancerBackendAddressPools for secondary ip configs.
+		// AzureCNI without overlay as pod networking type requires we configure additional secondary ips
+		// TODO: When MaxPods comes from the AKSNodeClass kubelet configuration, get the number of secondary 
+		// ips from the nodeclass instead of using the default
+		for i := 1; i < utils.DefaultMaxPods(networkPlugin, networkPluginMode); i++ {
+			nic.Properties.IPConfigurations = append(
+				nic.Properties.IPConfigurations,
+				&armnetwork.InterfaceIPConfiguration{
+					Name: lo.ToPtr(fmt.Sprintf("ipconfig-%d", i)),
+					Properties: &armnetwork.InterfaceIPConfigurationPropertiesFormat{
+						Primary: lo.ToPtr(false),
+						PrivateIPAllocationMethod: lo.ToPtr(armnetwork.IPAllocationMethodDynamic),
+						Subnet: &armnetwork.Subnet{
+							ID: &p.subnetID,
+						},
+					},
+				},
+			)
+		}
+	}
```

Comment:
Having kubeletconfiguration in the nodeclass would make this configuration for dynamic max pods easier. 

Otherwise we will end up adding some code to the launch template just to pass that configuration only to remove it a couple weeks later.